src/Microsoft.DotNet.Darc/src/Darc/Operations/CloneOperation.cs

```diff
@@ -28,25 +31,134 @@ public override async Task<int> ExecuteAsync()
+                        string repoPath = GetRepoDirectory(_options.ReposFolder, repo.RepoUri, repo.Commit);
+                        // the "master" folder, which continues to be linked to the .git directory
+                        string masterGitRepoPath = GetMasterGitRepoPath(_options.ReposFolder, repo.RepoUri);
+                        // the .gitdir that is shared among all repo-hashes (temporarily, before they are orphaned)
+                        string masterRepoGitDirPath = GetMasterGitDirPath(_options.GitDirFolder, repo.RepoUri);
+                        // used for the specific-commit version of the repo
+                        Local local;
+
+                        // Scenarios we handle: no/existing/orphaned master folder cross no/existing .gitdir
+                        await HandleMasterCopy(remoteFactory, repo.RepoUri, masterGitRepoPath, masterRepoGitDirPath, Logger);
+                        // if using the default .gitdir path, get that for use in the specific clone.
+                        if (masterRepoGitDirPath == null)
+                        {
+                            masterRepoGitDirPath = GetDefaultMasterGitDirPath(_options.ReposFolder, repo.RepoUri);
+                        }
+                        local = HandleRepoAtSpecificHash(repoPath, repo.Commit, masterRepoGitDirPath, Logger);
+
+                        Logger.LogDebug($"Starting to look for dependencies in {repoPath}");
+                        try
+                        {
+                            IEnumerable<DependencyDetail> deps = await local.GetDependenciesAsync();
+                            IEnumerable<DependencyDetail> filteredDeps = FilterToolsetDependencies(deps, _options.IncludeToolset);
+                            Logger.LogDebug($"Got {deps.Count()} dependencies and filtered to {filteredDeps.Count()} dependencies");
+                            filteredDeps.ForEach((d) =>
+                            {
+                                // e.g. arcade depends on previous versions of itself to build, so this would go on forever
+                                if (d.RepoUri == repo.RepoUri)
+                                {
+                                    Logger.LogDebug($"Skipping self-dependency in {repo.RepoUri} ({repo.Commit} => {d.Commit})");
+                                }
+                                else if (_options.IgnoredRepos.Any(r => r.Equals(d.RepoUri, StringComparison.OrdinalIgnoreCase)))
+                                {
+                                    Logger.LogDebug($"Skipping ignored repo {d.RepoUri} (at {d.Commit})");
+                                }
+                                else if (string.IsNullOrWhiteSpace(d.Commit))
+                                {
+                                    Logger.LogWarning($"Skipping dependency from {repo.RepoUri}@{repo.Commit} to {d.RepoUri}: Missing commit.");
```

Comment:
>LogWarning [](start = 43, length = 10)

Although probably this won't run as part of a build, most build definitions would treat warnings as errors failing the build.

Reply:
Honestly this seems like something that should be fixed if it comes up - it's probably a subscription that wasn't set up properly, right?  Or is there a valid use case were we won't have a hash for a Version.Details.xml dependency?

---
In reply to: [275515042](https://github.com/dotnet/arcade-services/pull/326#discussion_r275515042) [](ancestors = 275515042)